samtranslator/model/sam_resources.py

```diff
@@ -886,6 +889,11 @@ def _construct_version(
             # If SnapStart is enabled we want to publish a new version, to have the corresponding snapshot
             if function.SnapStart and function.SnapStart.get("ApplyOn", "None") != "None":
                 logical_dict.update({"SnapStart": function.SnapStart})
+            # We can't directly change AutoPublishAlias as that would be a breaking change, so we have to add this opt-in
+            # property that when set to true would change the lambda version whenever a property in the lambda function changes
+            if self.AutoPublishAliasAllProperties:
```

Comment:
Should this be supported in `Globals` as well?

Reply:
Not too sure, do we add every new property as `Globals` or only certain ones?

Reply:
It looks like `AutoPublishAlias` is, so probably want for this one as well: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification-template-anatomy-globals.html